src/features/views/components/MoveViewDialog.tsx

```diff
@@ -0,0 +1,266 @@
+      </DialogTitle>
+
+      <DialogContent>
+        <ZUIFuture future={itemsFuture}>
+          {(data) => {
+            const relevantItems = data.filter(
+              // This explicit typing should not be necessary, but it is. Don't know why.
+              (item): item is ViewBrowserFolderItem | ViewBrowserViewItem =>
+                item.type != 'back'
+            );
+            if (relevantItems.length == 0) {
+              return (
+                <List>
+                  <ListItem
+                    sx={{
+                      color: theme.palette.onSurface.disabled,
+                    }}
+                  >
+                    {messages.moveViewDialog.emptyFolder()}
+                  </ListItem>
+                </List>
+              );
+            }
+            return (
+              <List>
+                {relevantItems.map((item) => {
+                  const {
+                    data: { id },
+                    title,
+                    type,
+                  } = item;
+
+                  return (
+                    <ListItem
+                      key={`${type}-${id}`}
+                      onClick={
+                        type == 'folder' ? () => setViewedFolder(id) : undefined
+                      }
+                      sx={
+                        type == 'folder'
+                          ? { cursor: 'pointer' }
+                          : {
+                              color: theme.palette.onSurface.disabled,
+                            }
+                      }
+                    >
+                      <Box alignItems="center" display="flex" width="100%">
+                        <Box
+                          alignItems="flex-start"
+                          display="flex"
+                          marginRight={2}
+                        >
+                          <BrowserItemIcon item={item} />
+                        </Box>
+
+                        <Box
+                          alignItems="center"
+                          display="flex"
+                          flexGrow={1}
+                          marginRight={2}
+                        >
+                          {type == 'folder' ? (
+                            <Link color="inherit" underline="hover">
+                              {title}
+                            </Link>
+                          ) : (
+                            title
+                          )}
+                        </Box>
+                      </Box>
+                    </ListItem>
+                  );
+                })}
+              </List>
+            );
+          }}
+        </ZUIFuture>
+      </DialogContent>
+      <ZUIFuture future={viewTreeFuture}>
+        {(viewTree) => {
+          const tryingToMoveFolderIntoItself =
+            itemToMove.type == 'folder' &&
+            getAllParentFolderIds(viewedFolder, viewTree).some(
+              (id) => id == itemToMove.data.id
+            );
```

Comment:
I like this, but maybe there is an even more user-friendly way to prevent this, which is to simply not allow navigation into the folder? Like disabling clicking on the item somehow?

Reply:
That would be possible, but @Bagera and I settled on this solution when we searched for inspiration from similar services. I think he can argue the case better than I can 😄 But for now, I think I will leave this as is.